pkg/connectivity/pod-to-pod.go

```diff
@@ -25,12 +25,16 @@ func PodToPod(fromPod *v1.Pod, toPod *v1.Pod) common.Result {
 		// Check source Pod's namespace
 		namespace.IsInjectEnabled(client, fromPod.Namespace),
 		namespace.IsMonitoredBy(client, fromPod.Namespace, meshName),
-		pod.HasEnvoySidecar(fromPod),
+		pod.HasMinExpectedContainers(fromPod, 3),
+		pod.HasExpectedEnvoyImage(fromPod),
+		pod.HasProxyUUIDLabel(fromPod),
 
 		// Check destination Pod's namespace
 		namespace.IsInjectEnabled(client, toPod.Namespace),
 		namespace.IsMonitoredBy(client, toPod.Namespace, meshName),
-		pod.HasEnvoySidecar(toPod),
+		pod.HasMinExpectedContainers(toPod, 3),
```

Comment:
It seems to me that this should be 2 containers (Envoy + Payload).  And then maybe we can have `pod.HasInitContainer`

(And then on top of that we need to check if the init container sets up IPTables etc.)

Reply:
Going to do this in a follow up PR! Having some trouble with the HasInitContainer check failing